decorator.rb

```diff
@@ -0,0 +1,24 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(name)
+    super()
+    @nameable = name
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0..9]
+  end
+end
```

Comment:
- [ ] [OPTIONAL] i would highly recommend that you move your `CapitalizeDecorator` & `TrimmerDecorator` to a private files to keep your code clean & more readable 🙏